packages/preferences/src/browser/views/preference-tree-widget.tsx

```diff
@@ -140,15 +140,39 @@ export class PreferencesTreeWidget extends TreeWidget {
         if (!TreeNode.isVisible(node)) {
             return undefined;
         }
+
         const attributes = this.createNodeAttributes(node, props);
+        const visibleLeaves = this.calculateVisibleLeaves(node);
+        const printedNameWithVisibleChildren = node.name && visibleLeaves ? `${node.name} (${visibleLeaves})` : node.name;
+
         const content = <div className={TREE_NODE_CONTENT_CLASS}>
-            {this.renderExpansionToggle(node as Preference.Branch, props)}
-            {this.renderCaption(node, props)}
+            {this.renderExpansionToggle(node, props)}
+            {this.renderCaption({ ...node, name: printedNameWithVisibleChildren }, props)}
         </div>;
         return React.createElement('div', attributes, content);
     }
 
-    protected renderExpansionToggle(node: Preference.Branch, props: NodeProps): React.ReactNode {
+    protected calculateVisibleLeaves(node: Preference.TreeExtension): number {
+        let visibleLeaves = 0;
+        // The check for node.name prevents recursion at the level of `root`.
+        if (node.name && this.preferenceTreeProvider.isFiltered) {
+            if (node.children) {
+                node.children.forEach(child => {
+                    visibleLeaves += this.calculateVisibleLeaves(child);
+                });
+            }
+            if (node.leaves) {
+                node.leaves.forEach(leaf => {
+                    if (leaf.visible) {
+                        visibleLeaves++;
+                    };
+                })
```

Comment:
Compilation eror. Missing semi-colum at end of line 169

Here the error:
/home/lmcbout/theia/theia/packages/preferences/src/browser/views/preference-tree-widget.tsx
lerna ERR!   169:19  error  Missing semicolon  @typescript-eslint/semi
lerna ERR! 
lerna ERR! ✖ 1 problem (1 error, 0 warnings)
lerna ERR!   1 error and 0 warnings potentially fixable with the `--fix` option.